demo/midgard-node/src/workers/commit-block-header.ts

```diff
@@ -274,6 +284,7 @@ const inputData = workerData as WorkerInput;
 
 const program = pipe(
   wrapper(inputData),
+  Effect.provide(Database.layer),
```

Comment:
This is the only place `Database.layer` has been used, which doesn't _seem_ correct. Have you perhaps considered merging `database.ts` with `config.ts`? Do we even really need `database.ts`?